src/operator/quantization/quantize_v2-inl.h

```diff
@@ -41,7 +41,7 @@ struct QuantizeV2Param : public dmlc::Parameter<QuantizeV2Param> {
   int out_type;
   dmlc::optional<float> min_calib_range;
   dmlc::optional<float> max_calib_range;
-  dmlc::optional<bool> shifted;
+  dmlc::optional<bool> shifted_output;
```

Comment:
The dmlc::optional<T> is found a use for representing a value that may or may not be present. The std::optional was introduced in C++17, so it means that we can try to replace this one by a new one. Unfortunately, there is a bunch of branches have not supported C++17 yet.  An exceptation is the master branch that supports c++17, well, maybe is worth making an internal wrapper like this: [proposition]
```cpp
#define MXNET_HAS_OPTIONAL() 0
#if __cplusplus >= 201703L
#  ifdef __has_include
#    if __has_include(<optional>)
#      include <optional>
#      undef MXNET_HAS_OPTIONAL
#      define MXNET_HAS_OPTIONAL() 1
#    endif
#  endif
#endif

#if MXNET_HAS_OPTIONAL()
#warning "optional"
// Please use std::optional<T>
#else
#warning "no optional"
// Please use dmlc::optional<T>
#endif
```
What are the advantages of using it?

Reply:
I think v1.x is more strongly backward compatibility oriented. I think it would be better to keep such change only to master. dmlc::optional is used in a lot of places as well as other functions so it might be hard to justify rewriting parts of it.  I don't know what are advantages of either implementations. 
Also, in this PR I would prefer to keep consistency with what is already used in operators.